opentelemetry-api/src/opentelemetry/_metrics/__init__.py

```diff
@@ -129,17 +129,21 @@ def version(self):
     def schema_url(self):
         return self._schema_url
 
-    # FIXME check that the instrument name has not been used already
+    def _check_instrument_name(self, name):
+        if name in self._instrument_names:
+            raise Exception(f"Instrument name {name} has been used already")
+
+        self._instrument_names.add(name)
```

Comment:
This is should lock before updating the data structure

Reply:
Fixed :v: